mvpa2/misc/plot/scatter.py

```diff
@@ -0,0 +1,643 @@
+                axes[r, c] = pl.gca()
+
+        for d1 in xrange(0, n):
+            # we should unify the ranges of values displayed
+            ylim = np.min(d[d1]), np.max(d[d1])
+            for d2 in xrange(0, n):
+                if d1 == d2:
+                    continue
+                xlim = np.min(d[d2]), np.max(d[d2])
+                # only upper triangle
+                hint_opacity = kwargs.pop('hint_opacity', 0.9) if (d1==0 and d2==1) else 0
+                plot_scatter([d[d2], d[d1]],
+                             ax_scatter=axes[d1, d2],
+                             ax_hist_x=axes[d2, d2]
+                                       if (d2==d1+1 or (d1==1 and d2==0)) else None,
+                             ax_hist_y=None,
+                             bp_location='hist',
+                             hint_opacity=hint_opacity,
+                             xlim=xlim,
+                             ylim=ylim,
+                             **kwargs
+                             )
+                # adjust slightly
+                if not (d2==0 or (d1==0 and d2==1)): # not first column or in first row the first one
+                    # forget about y axis labels
+                    axes[d1, d2].yaxis.set_major_formatter(nullfmt)
+                if not (d1==n-1 or (d2==n-1 and d1==n-2)): # not first row or in first column in the first one
+                    # forget about y axis labels
+                    axes[d1, d2].xaxis.set_major_formatter(nullfmt)
+
+        if not (labels  in (None, [])):
+            assert len(labels) == n, "We should be provided all needed labels"
+            for d1,l in enumerate(labels):
+                axes[d1, 0].set_ylabel(l)
+            for d1,l in enumerate(labels):
+                axes[n-1,d1].set_xlabel(l)
+
+    else:
+        raise ValueError("Unknown style %s" % style)
+
+    return fig, axes
+
+
+def unique_path_parts(*paths):
+    paths_splits = [p.split(os.path.sep) for p in paths]
+    minlen = min([len(x) for x in paths_splits])
+    paths_split = np.array([p.split(os.path.sep, minlen-1) for p in paths]).T
+    paths_short = []
+    for p in paths_split:
+        if len(np.unique(p)) > 1:
+            # so do not match
+            paths_short.append(list(p))
+        else:
+            if not len(paths_short) or paths_short[-1][0] != '...':
+                paths_short.append(['...']*len(p))
+    ret = [os.path.sep.join(p) for p in np.array(paths_short).T]
+    return ret
+
+
+def _get_data(f):
+    """Adapter to load data from various formats
+    """
+    if f.endswith('.hdf5'):
+        from mvpa2.base.hdf5 import h5load
+        data = h5load(f).samples
+    else: #if f.endswith('.nii.gz') or f.endswith('.img') or f.endswith('.hdr'):
+        n = nb.load(f)
+        data = n.get_data()
+        # strip rudimentary 4th dimension
+        if len(data.shape) == 4 and data.shape[-1] == 1:
+            data = data[:, :, :, 0]
+    return data
+
+
+def plot_scatter_files(files,
```

Comment:
you could use e.g. those: mvpa2/data/haxby2001/sub001/anatomy/lowres00{1,2}.nii.gz ... probably openfmri thing could provide a concise path, but `pathjoin(pymvpa_dataroot, *('haxby2001/sub001/anatomy/lowres001.nii.gz'.split('/')))` should work ;)